susemanager-utils/susemanager-sls/modules/pillar/suma_minion.py

```diff
@@ -78,19 +114,156 @@ def ext_pillar(minion_id, pillar, *args):
     log.debug('Getting pillar data for the minion "{0}"'.format(minion_id))
     ret = {}
 
+    # Load the pillar from the legacy files
+    ret = load_legacy_pillars(minion_id, ret)
+
+    # Load the global pillar from DB
+    with _get_cursor() as cursor:
+        ret = load_global_pillars(cursor, ret)
+        ret = load_org_pillars(minion_id, cursor, ret)
+        group_formulas, ret = load_group_pillars(minion_id, cursor, ret)
+        system_formulas, ret= load_system_pillars(minion_id, cursor, ret)
+
+    # Including formulas into pillar data
+    try:
+        ret.update(formula_pillars(system_formulas, group_formulas, ret))
+    except Exception as error:
+        log.error('Error accessing formula pillar data: {message}'.format(message=str(error)))
```

Comment:
nitpick: In this case this shouldn't have a big impact on performance afaik since error messages are almost always logged and `str(error)` is not very expensive either, but in general we should make use of logging's lazy evaluation (strings are only formatted if they are logged).

For this call that would be 
```python
log.error('Error accessing formula pillar data: %s', error))